### PR TITLE
Adds basic discover page with search box

### LINKED ISSUE #148: Add /discover with search box

### PR DESCRIPTION
**Pull request by bartaz**
Fixes #148 

Adds basic MVP version of /discover page

screen shot 2017-11-28 at 13 36 41

### demo:
http://snapcraft.io-pr-155.run.demo.haus/discover

### CODE REVIEW

File: static/sass/styles.scss [hunk 1]
```diff
@@ -1,3 +1,7 @@
+// colour definitions
+$color-brand: #00302f; // TODO: this is a temprary brand colour
```
Comment (tbille) [inline static/sass/styles.scss]: This means that you want to wait before merging it?

Comment (Greggless) [reply static/sass/styles.scss]: We'll be able to merge using this colour, it's just temporary until we decide to change it at a later date.

But don't merge yet as it's in design review :)

Comment (tbille) [reply static/sass/styles.scss]: ok let's add labels to be sure no one merges it

**Comment by Greggless** [comment]
# Feedback
## Reviewed with @spencerbygraves & @kwm14 
- Match heading and strapline for sizing and max width to [snapcraft.io demo](http://snapcraft.io-pr-98.run.demo.haus/) (we looked at this and it appears that the h5s are different sizes. What could be causing this?)
- Search field to be 10-col
- We think that using the green background on the results pages works better (sorry!) 
- Can we put a redirect in for:
  - /search --> /discover
  - /search?q= --> /discover
- It feels a little odd to have the white space below the Green background, but a full length green background is too strong... 

Once again, speedy work! Thanks 😄

**Comment by bartaz** [comment]
@Greggless I think I need some clarifications here

> Match heading and strapline for sizing and max width to snapcraft.io demo (we looked at this and it appears that the h5s are different sizes. What could be causing this?)

Not sure what you mean about different h5 sizes, but I'll match the sizes and max-widths to what we started doing on the new home page.

> Search field to be 10-col

Should the text field be 10-col and button in the other 2-col, or should whole search (field + button) be 10 col wide?
 
> We think that using the green background on the results pages works better (sorry!)

Sure, wanted to propose it myself as it looks weird when green background changes to grey when jumping to search results.
Should the search page title be in the green background as well?

> Remove strapline on search results pages

Currently there is no strapline on search results pages, unless you mean `We’ve found some snaps for "test"` title? Should search results have no text on top at all, just search field?

> Can we put a redirect in for:
> /search --> /discover
> /search?q= --> /discover

Not sure what do you mean here?
You don't want to have `/search` URL at all? Should results be shown on `/discover` URL meaning `/discover` will serve both as discover page and search results page?

> It feels a little odd to have the white space below the Green background, but a full length green background is too strong...

Any suggestions how to deal with that?

File: static/sass/styles.scss (hunk 1 above)
Comment (bartaz) [reply static/sass/styles.scss]: It was just copied from new home page PR.
This is a brand colour that we use for now as the final snapcraft colour was not decided yet. But, as I understand we can use it until we have the final one.

**Comment by Greggless** [comment]
Ah, apologies for the ambiguity.

- The new homepage uses Lyubo's max width rules, so it'd be good to use those. 
When we (Spencer, Karl and I) looked at the straplines on both pages they appeared to be different sizes, but we don't know why as they are both H5s

- 10 col total for both search field and btn, please

- Search page title to be on the background too, that will replace 'Discover snaps for any Linux device'
- There will be no strapline on search results pages, only the H2 'We've found some snaps for <string>'
- For redirects, the idea was that if the search field is empty, or if the user enters the url snapcraft.io/search we will direct them to the Discover page. The behaviour currently is that if you hit the search button (taking the user to `/search?=`) then they get the search results page with the title 'Search snap store'. 

- For the white space issue, it was suggested that we use the footer grey (F7F7F7). But this will look weird when search results change colour to a white background, or return search results on a grey background. So I think for the MVP a white background is OK.

**Comment by bartaz** [comment]
@Greggless OK, thanks, I'm on it.

File: app.py [hunk 2]
```diff
@@ -168,13 +168,16 @@ def homepage():
...
+
+
 @app.route('/search')
 def search_snap():
     snap_searched = flask.request.args.get('q', default='', type=str)
     if(not snap_searched):
-        return flask.render_template(
-            'search.html'
-        )
+        return flask.redirect('/discover')
```
Comment (bartaz) [inline app.py]: @tbille @nottrobin Greg asked if we could redirect search with empty query to discover page. Do you think it's ok to do it this way?

Comment (tbille) [reply app.py]: It's okay for me

Comment (nottrobin) [reply app.py]: yeah okay

File: static/sass/styles.scss (hunk 1 above)
Comment (nottrobin) [reply static/sass/styles.scss]: @Greggless could you clarify - is it okay to merge with this colour? Or do we need a "Don't merge" label?

**Comment by Greggless** [comment]
@nottrobin @bartaz Yep, it's OK to merge 

:shipit:

**Comment by nottrobin** [comment]
@Greggless can you give it "Design +1"? Or should we just remove the "Design needed" label?

**Comment by Greggless** [comment]
Both 👍

**Comment by bartaz** [comment]
Notes from review meeting:

- more spacing between search results
- use green button

File: templates/discover.html [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{% extends "_layout.html" %}
+
+{% block title %}
+  Discover snaps for any Linux device — Linux software in the Snap Store
+{% endblock %}
+
+{% block content %}
+  <section id="main-content" class="p-strip--accent p-strip--image is-dark is-deep" style="background-image: url('https://assets.ubuntu.com/v1/0a575390-hero-background--edgeless.png'); background-size: 1440px 1440px; background-position: 50% 0;">
```
Comment (tbille) [inline templates/discover.html]: Same code for the search page. Move css to it's own file.

**Comment by Greggless** [comment]
@spencerbygraves @lyubomir-popov 

Here's the PR for sign off!

**Comment by lyubomir-popov** [comment]
@bartaz Should'n there be links in the header?

**Comment by bartaz** [comment]
@lyubomir-popov This page uses the same layout as snap details page (that doesn't have any navigation).

@Greggless do we need navigation here?

**Comment by Greggless** [comment]
Yes, this page will soon evolve into Phase 2 of the [directory](https://docs.google.com/a/canonical.com/document/d/11JP2Qzf_a1qe9U7ipaY03_Vd_0fyePnOWy1MsiZ3MnY/edit?usp=sharing) and we will need to link to other areas of snapcraft.io

**Comment by nottrobin** [comment]
@Greggless that was a confusing answer. Does this PR need to add a header to this page? I would have thought no - any navigation should be across the site and should therefore be a separate piece of work surely?

**Comment by Greggless** [comment]
@nottrobin @bartaz If it's easy enough to add with this PR then we should do it. If not, we can do it in a separate PR in this iteration or next, nay bother.

**Comment by nottrobin** [comment]
@Greggless could we please not add it to this PR?

I haven't seen a spec for it, so I doubt it will be trivial - do we even know exactly what links are going in it? In any case, pieces of work should be as atomic as possible, and so I see no reason at all to try to solve that separate issue in this PR.

**Comment by Greggless** [comment]
No problemo

FYI: Snap directory spec lives [here](https://docs.google.com/a/canonical.com/document/d/11JP2Qzf_a1qe9U7ipaY03_Vd_0fyePnOWy1MsiZ3MnY/edit?usp=sharing) with the navigation specc'd per phase

**Comment by lyubomir-popov** [comment]
@Greggless @carlaberkers I've added a Design +1 label, there are a few things I'd like to suggest but we've decided to postpone those for next iteration when this page will be revisited to add featured content etc.

**Pull request merged**